zcash_client_backend/proto/service.proto

```diff
@@ -8,18 +8,31 @@ option go_package = "lightwalletd/walletrpc";
 option swift_prefix = "";
 import "compact_formats.proto";
 
+// An identifier for a Zcash value pool.
+enum PoolType {
+    TRANSPARENT = 0;
+    SAPLING = 1;
+    ORCHARD = 2;
+}
+
 // A BlockID message contains identifiers to select a block: a height or a
 // hash. Specification by hash is not implemented, but may be in the future.
 message BlockID {
-     uint64 height = 1;
-     bytes hash = 2;
+    uint64 height = 1;
+    bytes hash = 2;
 }
 
 // BlockRange specifies a series of blocks from start to end inclusive.
 // Both BlockIDs must be heights; specification by hash is not yet supported.
+//
+// A server may optionally prune `CompactBlocks` returned to include only data
+// relevant to the requested pool types. If no pool types are specified, the
+// server should default to the legacy behavior of returning only data relevant
+// to the shielded (Sapling and Orchard) pools.
 message BlockRange {
     BlockID start = 1;
     BlockID end = 2;
+    repeated PoolType poolTypes = 3;
```

Comment:
`BlockRange` is also used in `TransparentAddressBlockFilter`, where it does not make sense to restrict the pool types. We should document that `poolTypes` MUST NOT be present when it is used in `TransparentAddressBlockFilter` (which is the argument type to `rpc GetTaddressTxids`).